input-profile.js

```diff
@@ -0,0 +1,249 @@
+
+        let children = obj.children;
+        for (let i = 0; i < children.length; ++i) {
+            this.getComponents(children[i]);
+        }
+    }
+
+    setComponentsActive(active) {
+        const comps = this.components;
+        for (let i = 0; i < comps.length; ++i) {
+            comps[i].active = active;
+        }
+    }
+
+    loadProfileJson(xrInputSource) {
+        /** load Profile json */
+        /** Note: if you are providing a custom ProfileJSON provide its path in the
+         * customProfileFolder property of the input-profile component.
+         */
+        this.profile = this.customProfileFolder || this.path + xrInputSource.profiles[0];
+        this.url = this.profile + '/profile.json';
+        if (Object.keys(this.ProfileJSON).length > 0) {
+            this.loadModel(xrInputSource);
+        } else {
+            fetch(this.url)
+                .then((res) => res.json())
+                .then((out) => {
+                    this.ProfileJSON = out;
+                    this._profileJsonLoaded = true;
+                    this.loadModel(xrInputSource);
+                })
+                .catch((error) => {
+                    console.log('couldnot read profile.json ' + this.url);
+                });
+        }
+    }
+
+    loadModel(xrInputSource) {
+        if (this.handedness === xrInputSource.handedness) {
+            this.gamepad = xrInputSource.gamepad;
+            /** load controllerModel **/
+
+            if (this.modelLoaded) return;
+
+            if (this.preSpawnedController !== null) {
+                this.controllerObject = this.preSpawnedController;
+                this.controllerObject.parent = this.object;
+                this.controllerObject.setTranslationLocal([0, 0, 0]);
+                this.getGamepadObjectsFromProfile(this.ProfileJSON, this.controllerObject);
+                this.modelLoaded = true;
+                console.log('Loaded ' + this.handedness + '  controller');
+            } else {
+                const assetPath = this.profile + '/' + this.handedness + '.glb';
+                this.engine.scene
+                    .append(assetPath)
+                    .then((obj) => {
+                        this.controllerObject = obj;
+                        this.components = [];
+                        this.getComponents(obj);
+                        this.controllerObject.parent = this.object;
+                        this.controllerObject.setTranslationLocal([0, 0, 0]);
+                        this.getGamepadObjectsFromProfile(
+                            this.ProfileJSON,
+                            this.controllerObject
+                        );
+                        this.modelLoaded = true;
+                        console.log('Loaded ' + this.handedness + '  controller');
+                    })
+                    .catch(console.error);
+            }
+        }
+    }
+
+    getGamepadObjectsFromProfile(profile, obj) {
+        //initialise components
```

Comment:
We use `/*` for comments, and a space between the beginning of the comment and the first word would be amazing